### PR TITLE
rgw: fix memory leak in delete_obj_aio

### PR DESCRIPTION
**Pull request by wangzhengyong**
Signed-off-by: wangzhengyong@cmss.chinamobile.com

### CODE REVIEW

**Comment by prallabh** [comment]
@wangzhengyong You have done it for delete_raw_obj_aio(), you may want to fix a similar one in RGWRados::delete_obj_aio() further down in the same file.

**Comment by wangzhengyong** [comment]
@prallabh done

**Comment by wangzhengyong** [comment]
please retest

File: src/rgw/rgw_rados.cc [hunk 1]
```diff
@@ -12956,6 +12956,7 @@ int RGWRados::delete_raw_obj_aio(const rgw_raw_obj& obj, list<librados::AioCompl
   ret = ref.ioctx.aio_operate(ref.oid, c, &op);
   if (ret < 0) {
     lderr(cct) << "ERROR: AioOperate failed with ret=" << ret << dendl;
+    delete c;
```
Comment (yehudasa) [inline src/rgw/rgw_rados.cc]: @wangzhengyong should call c->release() instead.

File: src/rgw/rgw_rados.cc [hunk 2]
```diff
@@ -12994,6 +12995,7 @@ int RGWRados::delete_obj_aio(const rgw_obj& obj,
...
```
Comment (yehudasa) [inline src/rgw/rgw_rados.cc]: @wangzhengyong same here

Comment (wangzhengyong) [inline src/rgw/rgw_rados.cc]: ok，this right

**Comment by cbodley** [comment]
passed rgw suite: http://pulpito.ceph.com/cbodley-2017-04-03_16:14:27-rgw-wip-cbodley-testing---basic-mira

**Comment by cbodley** [comment]
jenkins test this please

**Pull request merged**